Source/Core/Common/FileUtil.cpp

```diff
@@ -651,15 +651,14 @@ void CopyDir(const std::string &source_path, const std::string &dest_path)
 // Returns the current directory
 std::string GetCurrentDir()
 {
-	char *dir;
-	// Get the current working directory (getcwd uses malloc)
-	if (!(dir = __getcwd(nullptr, 0)))
+	char* dir=(char*) malloc(PATH_MAX*sizeof(char));
+	if (!(getcwd(dir, PATH_MAX)))
 	{
 		ERROR_LOG(COMMON, "GetCurrentDirectory failed: %s",
 				GetLastErrorMsg().c_str());
 		return nullptr;
 	}
-	std::string strDir = dir;
+	std::string strDir = std::string(dir);
```

Comment:
To me a c string and a c++ string are two different types, so it does not make sense to be able to set a c++ string to the pointer for a c string.  As a result, using that constructor is much more correct for me.